xbmc/cores/VideoPlayer/Buffers/VideoBufferDMA.cpp

```diff
@@ -119,20 +119,25 @@ bool CVideoBufferDMA::Alloc()
 
 void CVideoBufferDMA::Export(AVFrame* frame, uint32_t width, uint32_t height)
 {
-  m_planes = av_pix_fmt_count_planes(static_cast<AVPixelFormat>(frame->format));
+  AVPixelFormat pix_fmt = static_cast<AVPixelFormat>(frame->format);
+  m_planes = av_pix_fmt_count_planes(pix_fmt);
+  int h_shift, v_shift;
 
-  if (m_planes < 2)
-    throw std::runtime_error(
-        "non-planar formats not supported: " +
-        std::string(av_get_pix_fmt_name(static_cast<AVPixelFormat>(frame->format))));
+  if (av_pix_fmt_get_chroma_sub_sample(pix_fmt, &h_shift, &v_shift))
+    throw std::runtime_error("unable to determine chroma_sub_sample: " +
+                             std::string(av_get_pix_fmt_name(pix_fmt)));
+
+  if (m_planes < 2 || m_planes > 3)
+    throw std::runtime_error("only 2 or 3 plane formats supported: " +
+                             std::string(av_get_pix_fmt_name(pix_fmt)));
 
   for (uint32_t plane = 0; plane < m_planes; plane++)
-  {
```

Comment:
I'd rather keep these braces for clarity please.

Reply:
Updated